common/functions/src/scalars/strings/oct.rs

```diff
@@ -0,0 +1,146 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt;
+
+use common_datavalues::prelude::*;
+use common_exception::ErrorCode;
+use common_exception::Result;
+
+use crate::scalars::function_factory::FunctionDescription;
+use crate::scalars::function_factory::FunctionFeatures;
+use crate::scalars::Function;
+
+trait OctString {
+    fn oct_string(self) -> String;
+}
+
+macro_rules! impl_oct_string_for_signed {
+    ($($ty:ty),*) => {
+        $(
+        impl OctString for $ty {
+            fn oct_string(self) -> String {
+                if self < 0 {
+                    format!("-0{:o}", self.unsigned_abs())
+                } else if self == 0 {
+                    "0".to_string()
+                } else {
+                    format!("0{:o}", self)
+                }
+            }
+        }
+        )*
+    };
+}
+
+impl_oct_string_for_signed!(i8, i16, i32, i64);
+
+macro_rules! impl_oct_string_for_unsigned {
+    ($($ty:ty),*) => {
+        $(
+        impl OctString for $ty {
+            fn oct_string(self) -> String {
+                if self == 0 {
+                    "0".to_string()
+                } else {
+                    format!("0{:o}", self)
+                }
+            }
+        }
+        )*
+    };
+}
+
+impl_oct_string_for_unsigned!(u8, u16, u32, u64);
+
+macro_rules! impl_oct_function {
+    ($column:expr, $ty:ident) => {{
+        let series = $column.column().to_minimal_array()?;
+        let primitive_array = series.$ty()?;
+
+        let mut string_array = StringArrayBuilder::with_capacity($column.column().len());
+        for value in primitive_array {
+            string_array.append_option(value.map(|n| n.oct_string()));
+        }
+        let column: DataColumn = string_array.finish().into();
+        Ok(column.resize_constant($column.column().len()))
+    }};
+}
+
+#[derive(Clone)]
+pub struct OctFunction {
+    _display_name: String,
+}
+
+impl OctFunction {
+    pub fn try_create(display_name: &str) -> Result<Box<dyn Function>> {
+        Ok(Box::new(OctFunction {
+            _display_name: display_name.to_string(),
+        }))
+    }
+
+    pub fn desc() -> FunctionDescription {
+        FunctionDescription::creator(Box::new(Self::try_create))
+            .features(FunctionFeatures::default().deterministic())
+    }
+}
+
+impl Function for OctFunction {
+    fn name(&self) -> &str {
+        "oct"
+    }
+
+    fn num_arguments(&self) -> usize {
+        1
+    }
+
+    fn variadic_arguments(&self) -> Option<(usize, usize)> {
+        None
+    }
+
+    fn return_type(&self, args: &[DataType]) -> Result<DataType> {
+        if !args[0].is_integer() {
```

Comment:
args[0] can be DataType::null.

if args[0] is String, we can cast it into  Int64 using `cast_with_type`

MySQL support explicit cast:

select oct('56');
select oct('56a');
select oct(null);